pkg/pod/creds_init.go

```diff
@@ -43,8 +44,10 @@ const credsInitHomeMountPrefix = "tekton-creds-init-home"
 // caller. If no matching annotated secrets are found, nil lists with a
 // nil error are returned.
 func credsInit(serviceAccountName, namespace string, kubeclient kubernetes.Interface) ([]string, []corev1.Volume, []corev1.VolumeMount, error) {
+	// service account if not specified in pipeline/task spec, read it from the ConfigMap
+	// and defaults to `default` if its missing from the ConfigMap as well
 	if serviceAccountName == "" {
-		serviceAccountName = "default"
+		serviceAccountName = config.DefaultServiceAccountValue
```

Comment:
It's interesting that we force a service account to be specified, so we do not allow taskruns and pipelineruns to rely on the default for the namespace...  
With the changes from this PR in place, the only case where serviceAccountName was empty here would be if it was sent to "" in the configmap or in the pipelinerun/taskrun - so I wonder if should instead fail or at least log a warning?

Reply:
nope, it's perfectly fine to have service account name set to "". This change is avoiding hard coded `default` string here and instead reading from constant.

Reply:
What I meant is that we do not allow serviceAccount to be an empty string - we did not before and we still not do. My concern is that if the service account is set to empty string in the config map, we do not honour that, we override it instead with `config.DefaultServiceAccountValue` (which is the string "default"). This is not a change introduced in this PR, it's something that existed before, but I think it would be worth (perhaps in a different PR) to add a warning or something so the user is aware that the service account name has been overwritten compared to config.

Reply:
I see what you mean, warning should work. Will create a separate PR. Warn just the service account or include other defaults as well happening in [default.go](https://github.com/tektoncd/pipeline/blob/master/pkg/apis/config/default.go)?